src/librustc/middle/infer/combine.rs

```diff
@@ -278,7 +278,7 @@ impl<'a, 'tcx> CombineFields<'a, 'tcx> {
         };
         let u = ty.fold_with(&mut generalize);
         if generalize.cycle_detected {
-            Err(ty::terr_cyclic_ty)
```

Comment:
Yeah I wanted to do it, but wasn't sure how much pain it was to fix. I'll bite the bullet.